components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/HttpUnRegistrationResponseFactoryTest.java

```diff
@@ -0,0 +1,200 @@
+
+import javax.servlet.http.HttpServletResponse;
+
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Unit test covering HttpUnregistrationResponseFactory
+ */
+@PrepareForTest({HttpUnregistrationResponseFactory.class, HttpIdentityResponse.HttpIdentityResponseBuilder.class})
+public class HttpUnRegistrationResponseFactoryTest extends PowerMockTestCase {
+
+    private UnregistrationResponse mockUnregistrationResponse;
+    private HttpUnregistrationResponseFactory httpUnregistrationResponseFactory;
+    private HttpIdentityResponse.HttpIdentityResponseBuilder mockHttpIdentityResponseBuilder;
+
+    @BeforeMethod
+    private void setUp() {
+
+        mockUnregistrationResponse = mock(UnregistrationResponse.class);
+        httpUnregistrationResponseFactory = new HttpUnregistrationResponseFactory();
+    }
+
+    @Test
+    public void testCanHandle() throws Exception {
+
+        IdentityResponse identityResponse = mock(IdentityResponse.class);
+        assertFalse(httpUnregistrationResponseFactory.canHandle(identityResponse));
+    }
+
+    @Test
+    public void testCanHandleFailed() {
+
+        assertTrue(httpUnregistrationResponseFactory.canHandle(mockUnregistrationResponse));
+    }
+
+    @DataProvider(name = "exceptionInstanceProvider")
+    public Object[][] getExceptionInstanceType() {
+
+        return new Object[][]{
+                {new UnRegistrationException("")},
+                {new UnRegistrationException("", "dummyMessage")},
+                {new UnRegistrationException("", new Throwable())},
+                {new UnRegistrationException("", "dummyMessage", new Throwable())}
+        };
+    }
+
+    @Test(dataProvider = "exceptionInstanceProvider")
+    public void testCanHandleException(Object exception) {
+
+        Assert.assertTrue(httpUnregistrationResponseFactory.canHandle((UnRegistrationException) exception));
+    }
+
+    @Test
+    public void testCanHandleExceptionFailed() {
+
+        Assert.assertFalse(httpUnregistrationResponseFactory.canHandle(new FrameworkException("")));
+    }
+
+    @DataProvider(name = "instanceDataProvider")
+    public Object[][] getInstanceData() {
+
+        return new Object[][]{
+                {mockHttpIdentityResponseBuilder, mockUnregistrationResponse},
+                {null, mockUnregistrationResponse}
+        };
+    }
+
+    @Test(dataProvider = "instanceDataProvider")
+    public void testCreate(Object builder, Object unregistrationResponse) throws Exception {
```

Comment:
Do we need to pass the `unregistrationResponse` object as an argument, since it is a class variable?